### DISCUSSION TITLE
Trouble with class files when uploading Ultima Layout 7.0.0 to Artifactory

### DISCUSSION THREAD

**Discussion by mertcanmutlu13** [primefaces/community#1365] [PrimeFaces Templates] [answered]
Hello;

We are using Ultima Layout v7.0.0. As the DevOps team, we download the zip file from https://www.primefaces.org/store/ and upload it to our local repository (Artifactory). There haven't been any issues with other layouts (For example Apollo) we have used before. However, when I upload the jar file to Artifactory in Ultima, it doesn't contain class files. Our developers are unable to use it this way, so they end up taking the zip file from us and embedding it into the code. How can I deploy to the repository in a way that includes the class files?

Secondly, if possible, do you have a usable repository address for this layout? If so, we could mirror the address from your repository to ours and automatically get the latest version. This way, the problem mentioned above would be resolved automatically.

Thank you.

**Comment by NicolaIsotta** [comment]
Have you tried buiding the jar yourself?
`mvn clean install -P distribute`

**Reply by melloware** [reply, marked as answer]
or to Artifactory `mvn clean deploy -P distribute`

**Comment by mertcanmutlu13** [comment]
@NicolaIsotta @melloware Thank you, we were able to use it this way. So, is it possible to mirror the address I mentioned in the second scenario? If so, how can we do it?

**Reply by melloware** [reply]
as for as I know PrimeTek does not make mirror available of the Premium Themes.  This would expose all premium themes to the world.
